libs/template/templates/default-python/template/{{.project_name}}/resources/{{.project_name}}.pipeline.yml.tmpl

```diff
@@ -10,6 +14,9 @@ resources:
       catalog: {{default_catalog}}
       {{- end}}
       target: {{.project_name}}_${bundle.target}
+      {{- if $with_serverless }}
+      serverless: true
```

Comment:
There may be another special case here that should be tested:

I believe serverless pipelines always make use of unity catalog. And serverless compute for serverless only works if unity catalog is enabled on a workspace.

I'm not sure what the "check serverless available" API responds for workspaces without a catalog attached, but it would be good to check that. If the API says "yes" then we likely need to use the `(eq default_catalog "")` check from above. (That check always succeeds on workspaces _without_ a catalog. But I'm not 100% sure if will always fail on a workspace _with_ a catalog, so I'd rather not use that check if we don't need it.)
*

Reply:
Ah, so you're saying that 'catalog:' must always be present in serverless case? 

Does it means that with_serverless should be set to 'with_serverless := with_serverless and (catalog != "")`?